modules/nginx/manifests/restart.pp

```diff
@@ -1,7 +1,7 @@
 # FIXME: This class needs better documentation as per https://docs.puppetlabs.com/guides/style_guide.html#puppet-doc
 class nginx::restart {
 
-  exec { '/etc/init.d/nginx configtest && /etc/init.d/nginx restart':
+  exec { '/etc/init.d/nginx configtest && killall nginx && /etc/init.d/nginx restart':
```

Comment:
:) Can we please make this sth a bit more graceful, like: 
```
 '/etc/init.d/nginx configtest && { /etc/init.d/nginx restart || { pkill nginx; /etc/init.d/nginx start; } }'
```
I do not think the above will work directly - maybe we can use shellquote to create a $command variable or something similar..